manifests/groups.pp

```diff
@@ -0,0 +1,63 @@
+# This class takes two `Hashes` of group resources, one to remove, and one to
+# install.
+#
+# After the entire puppet catalog has been compiled, it will process both lists
+# and, for any resource that is not already defined in the catalog, it will
+# take the appropriate action.
+#
+# An exception will be raised if you list the same group in both lists.
+#
+# @param $remove
+#   A list of groups to remove.
+#
+#   * A `Hash` can be used to add extra attributes for the group, but the
+#     `ensure` attribute will be overwritten if it is included.
+#
+# @param $install
+#   A list of groups to install.
+#
+#   * A `Hash` can be used to add extra attributes for the group, but the
+#     `ensure` attribute will always be set to `$group_ensure`.
+#
+# @param $default_options
+#   A `Hash` of options to apply to all groups (both remove and install.
+#   If ensure is entered in these options it will be overwritten.
+#
+#   * These options may be anything that a Puppet `Group` resource can
+#     normally accept.
+#
+# @param $mode
+#   @see `deferred_resources::mode`
+#
+# @param $log_level
+#   @see `deferred_resources::log_level`
+#
+class deferred_resources::groups (
+  Variant[Hash, Array[String[1]]]      $remove          = {},
+  Variant[Hash, Array[String[1]]]      $install         = {},
```

Comment:
What is the use case for a Hash for $remove and $install ?  Why isn't an Array sufficient?

Reply:
@lnemsick-simp For removal, an array is sufficient. For creation, you may want to specify attributes such as the UID/GID, etc...

Will switch `remove` to just be an Array.